extension/js/common/message-renderer.ts

```diff
@@ -0,0 +1,782 @@
+/* ©️ 2016 - present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com */
+
+'use strict';
+
+import { GmailParser, GmailRes } from './api/email-provider/gmail/gmail-parser.js';
+import { PubLookup } from './api/pub-lookup.js';
+import { ClientConfiguration } from './client-configuration.js';
+import { Attachment, Attachment$treatAs, TransferableAttachment } from './core/attachment.js';
+import { Buf } from './core/buf.js';
+import { CID_PATTERN, Dict, Str, Value } from './core/common.js';
+import { KeyUtil } from './core/crypto/key.js';
+import { DecryptErrTypes, DecryptResult, FormatError, MsgUtil, VerifyRes } from './core/crypto/pgp/msg-util.js';
+import { PgpArmor } from './core/crypto/pgp/pgp-armor.js';
+import { Mime, MimeContent, MimeProccesedMsg } from './core/mime.js';
+import { MsgBlockParser } from './core/msg-block-parser.js';
+import { MsgBlock } from './core/msg-block.js';
+import { Lang } from './lang.js';
+import { Catch } from './platform/catch.js';
+import { AcctStore } from './platform/store/acct-store.js';
+import { ContactStore } from './platform/store/contact-store.js';
+import { KeyStore } from './platform/store/key-store.js';
+import { PassphraseStore } from './platform/store/passphrase-store.js';
+import { Xss } from './platform/xss.js';
+import { RelayManager } from './relay-manager.js';
+import { RenderInterface, RenderInterfaceBase } from './render-interface.js';
+import { MessageInfo, PrintMailInfo } from './render-message.js';
+import { saveFetchedPubkeysIfNewerThanInStorage } from './shared.js';
+import { XssSafeFactory } from './xss-safe-factory.js';
+import * as DOMPurify from 'dompurify';
+import { Downloader, ProcessedMessage } from './downloader.js';
+import { JQueryEl, LoaderContextBindInterface, LoaderContextBindNow, LoaderContextInterface } from './loader-context-interface.js';
+import { Gmail } from './api/email-provider/gmail/gmail.js';
+import { ApiErr, AjaxErr } from './api/shared/api-error.js';
+import { isCustomerUrlFesUsed } from './helpers.js';
+
+export type ProccesedMsg = MimeProccesedMsg;
+
+export class MessageRenderer {
+  public readonly downloader: Downloader;
+  private constructor(
+    private readonly acctEmail: string,
+    private readonly gmail: Gmail,
```

Comment:
I wonder if we should avoid using `Gmail` object directly, but delegate all download operations to `Downloader`.
`Downloader` provides some extra features e.g. caching.

Reply:
It'll be nice to have all download logic in one `Downloader` class, can be improvement for another PR